common/schema/definition.go

```diff
@@ -377,10 +378,10 @@ END`,
 					`concat(toString(ICMPv6Type), '/', toString(ICMPv6Code))), '')`,
 			},
 		},
-	}.finalize()
+	}.finalize(ColumnLast)
 }
 
-func (schema Schema) finalize() Schema {
+func (schema Schema) finalize(lastKey ColumnKey) Schema {
```

Comment:
Isn't the information included in the schema?

Reply:
ColumnLast is a const variable pointing to the last hardcoded dimension. With the "flexible", custom-dict based dimensions, the last column identifiert needs to move, thats why I made it flexible with that argument.

Shall we move the actual last key into the schema struct, or use another option?

Reply:
`ColumnLast` was a very small optimization in `finalize()`. Instead of modifying a lot of call sites, just keep the current signature. You could replace `ColumnLast` by `len(schema.columns)` or just use `append()` to update `schema.ColumnIndex`.

Reply:
Taking only the len does not work due to disabled columns, and I'd try not to mess with the index outside finalize.
I've added the schema.dynamicColumns variable, clearly indicating how many dynamic columns are added after the ColumnLast.

Reply:
With `len(schema.columns)` in common/schema/definition.go:507, I've got 
```
 runtime error: index out of range
```
 
What implementation do you mean exactly with append? Directly appending to schema.columnIndex from the column creating loop in schema/root.go?
 
Another idea would be to provide a `schema.appendColumn(Column)` (or similar) method, encapsulating everything that is needed to add a new column to an existing schema.

How shall we continue with this inside this PR?

Reply:
Yes, instead of building the array at the right size from the beginning, use an empty one and append to it.

Reply:
Where would you set/append the columnIndex list in this case? If I append inside the New method in root.go, it would be overwritten by the last finalize in https://github.com/akvorado/akvorado/pull/840/files#diff-08a5918f6f118e79e5971a4585ae6ea800df9e6d7527eb29bdba1ca5ce3dc623R147 as far as I can tell.

Reply:
I lured you in the wrong direction as I didn't realize this was an inverted index. Instead, I did that: d1cef41849e7.